packages/admin-api-client/src/admin-api-client.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  createGraphQLClient,
+  CustomFetchApi,
+  RequestParams as GQLClientRequestParams,
+  getCurrentSupportedApiVersions,
+  validateApiVersion,
+} from "@shopify/graphql-client";
+
+import {
+  AdminApiClient,
+  AdminApiClientConfig,
+  AdminApiClientRequestOptions,
+} from "./types";
+import {
+  DEFAULT_CONTENT_TYPE,
+  ACCESS_TOKEN_HEADER,
+  CLIENT,
+  DEFAULT_CLIENT_VERSION,
+} from "./constants";
+import {
+  validateRequiredStoreDomain,
+  validateRequiredAccessToken,
+  validateServerSideUsage,
+} from "./validations";
+
+const httpRegEx = new RegExp("^(https?:)?//");
+
+export function createAdminApiClient({
+  storeDomain,
+  apiVersion,
+  accessToken,
+  userAgentPrefix,
+  customFetchApi: clientFetchApi,
+}: {
+  storeDomain: string;
+  apiVersion: string;
+  accessToken: string;
+  userAgentPrefix?: string;
+  customFetchApi?: CustomFetchApi;
+}): AdminApiClient {
+  const currentSupportedApiVersions = getCurrentSupportedApiVersions();
+
+  validateServerSideUsage();
+  validateRequiredStoreDomain(storeDomain);
+  validateApiVersion({
+    client: CLIENT,
+    currentSupportedApiVersions,
+    apiVersion,
+  });
+  validateRequiredAccessToken(accessToken);
+
+  const trimmedStoreDomain = storeDomain.trim();
+  const cleanedStoreDomain = httpRegEx.test(trimmedStoreDomain)
+    ? trimmedStoreDomain.substring(trimmedStoreDomain.indexOf("//") + 2)
+    : trimmedStoreDomain;
```

Comment:
The `validateDomainAndGetStoreUrl()` util can help you validate and get the store URL